### ISSUE TITLE
IE Roadmap

### ISSUE THREAD

**Issue by jpfairbanks** [jpfairbanks/SemanticModels.jl#9]
We need to have a plan for the information extraction tasks. Lets keep this thread open for discussions on the planning what to do for IE. For concrete action items, open new issues and pull requests. Then tag the new issue in the main list.

Start with https://github.com/epirecipes/epicookbook "the cookbook"

===

@scottagt's Note: Updated 12-11 @ 11am

- [ ] Task 1 - Feature Design and Development
 - [ ] Features for Papers
   - [ ] Lexical Token Patterns
   - [ ] POS Parses
   - [ ] POS Embeddings
   - [ ] Others?
 - [x] Features for Paper Script Extraction
   - [x] Source Code Parse Tree
   - [x] Knowledge Graph Sub-trees
   - [x] Linters
   - [x] Others?
- [ ] Task 2 - Feature Extraction
  - [ ] Implement Feature Extractors for Papers
    - [ ] Eidos Rules
  - [x] Implement Feature Extractors for Scripts
    - [x] Can we use Eidos for this?
- [x] Task 3 - Annotation (starting with Eidos, waiting on this area for now)
- [ ] Task 4 - Model Building (starting with Eidos, waiting on this area for now)

===

- [ ] Extract text from sources
  - [x] Markdown files
  - [ ] Equations
  - [x] Notebooks
- [ ] Build knowledge graph from text
  - [x] Identify concepts
  - [x] Identify relations between concepts
  - [ ] Retain original source contexts
- [x] Associate extracted concepts to variable and function names from code. 

@scottagt, @crherlihy do you see any other tasks that should be on the roadmap for this?

**Comment by scottagt** [comment]
Hi @jpfairbanks, in case you're catching up on messages, wanted to make you aware that I wrote you about this last week and and an email today, about putting together a plan and tasks for IE. To summarize, @crherlihy and I have been putting together a list of IE tasks and a conceptual map that could be useful for contextualizing the other areas as well. I'd like to discuss and revise those and ultimately merge into this discussion thread tomorrow.

**Comment by jpfairbanks** [comment]
Scott, my latest email from you says

> Based on our chat Friday afternoon, I’m going to update the proposed IE Task list (to discuss tomorrow) to reflect leveraging Eidos platform among any others and what the efforts are to write new Odin style grammars, etc. Also, using that code from the Epireceipes repo, we’ll write up what different libraries are being imported, etc, in lieu of all the other libraries we had been reviewing for the earlier deliverable.

Please post that update to this issue before our meeting today so that I can be prepared for the discussion. We only have 5 working weeks until the next release is due, so we need to make the most of our weekly meetings.

**Comment by jpfairbanks** [comment]
Sorry, I was writing for the perspective that you were reading it in the morning so "today" referred to this morning 12/11. Thanks for the update, I read the word doc this morning.

**Comment by scottagt** [comment]
Oh no, it’s okay, I read today to be 12/11 when I replied a few minutes ago. I sent you a response email last night to your original comment referring to the earlier emails when you created the github issue but, your comment at 7:18 was after I sent you that email and updated word doc. I’m assuming perhaps you had just wrote on github before reading my email from last night. :) At any rate, I’ll update this issue in GitHub using those updates and from our discussion today.
________________________________

**Comment by jpfairbanks** [comment]
This new linear algebra textbook just hit the julia community influencer network. http://vmls-book.stanford.edu/vmls-julia-companion.pdf Chapter 5 has an example of a linear dynamical system. Guess what domain they chose...

**Comment by jpfairbanks** [comment]
Closed by #61
